core/src/main/java/io/grpc/internal/ClientCallImpl.java

```diff
@@ -63,6 +63,7 @@
   private final CallOptions callOptions;
   private ClientStream stream;
   private volatile ScheduledFuture<?> deadlineCancellationFuture;
+  private volatile boolean inOnReady;
```

Comment:
Switched to use thread ids to track causality of the flush. This eliminates the need for a volatile too